signer/src/transaction_signer.rs

```diff
@@ -903,7 +903,7 @@ mod tests {
 
         testing::transaction_signer::TestEnvironment {
             context,
-            context_window: 3,
+            context_window: 6,
```

Comment:
This is a magic number: higher we may fail the `num_expected_decisions` checks in `assert_should_store_decisions_received_from_other_signers` (because of context larger than chain length, so we expect more decision that we can get), lower we may get no withdrawal requests in the intersection between the stacks canonical chain and the context window.

Reply:
Ahhh I think I understand.

Reply:
Well, we have a known frailty with the generation of data for some of these tests. We can choose to fix them in some other PR.